### PR TITLE
docs: add content for debugging tools

### PR DESCRIPTION
**Pull request by arjunattam**

### CODE REVIEW

File: docs/debug.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+new debugger does not require a `launch.json` configuration. To use this:
+
+1. Enable the preview debugger
+    * Open JSON settings and add `"debug.javascript.usePreview": true`
+    * Open settings UI and enable the `Debug › JavaScript: Use Preview` setting
+
+1. Set a breakpoint in VS Code
+    * Use the `debugger` keyword or set a breakpoint in the VS Code UI
+
+1. Run your Node.js script from the terminal
```
Comment (dgozman) [inline docs/debug.md:54]: Shouldn't this be a special debugging terminal?

File: docs/debug.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+* **Run in headful**: With PWDEBUG, browsers always launch in headful mode
+* **Disables timeout**: PWDEBUG sets timeout to 0 (= no timeout)
+* **Preserve DevTools preferences**: When used with `devtools: true`, PWDEBUG
+  preserves the docked/undocked state of Chrome DevTools
+
+### Debugging Selectors
+
+PWDEBUG configures a `playwright` object in the browser, which can be used to
+inspect with [Playwright selectors](selectors.md), including the text selector
```
Comment (dgozman) [inline docs/debug.md:104]: `inspect with Playwright selectors` looks awkward to me 😄

File: docs/debug.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+or composite selectors. To use this:
+
+1. Open developer tools console during execution
```
Comment (dgozman) [inline docs/debug.md:107]: ...., for example when paused on a breakpoint in the Playwright script.

File: docs/debug.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+### Evaluate Source Maps
+
+PWDEBUG also enables source maps for [`page.evaluate` executions](core-concepts.md#evaluation).
+This improves the debugging experience for JavaScript executions in the page context.
+
+<a href="https://user-images.githubusercontent.com/284612/86857568-a6c63100-c073-11ea-82a4-bfd531a4ec87.png"><img src="https://user-images.githubusercontent.com/284612/86857568-a6c63100-c073-11ea-82a4-bfd531a4ec87.png" width="500" alt="Highlight selectors"></a>
+
+## Verbose API logs
+
+Playwright supports verbose logging with the `DEBUG` environment variable. This
```
Comment (dgozman) [inline docs/debug.md:127]: `This outputs` sentence is unfinished.

**Review by dgozman — APPROVED** [review]
Wonderful!

File: docs/debug.md (hunk 1 above)
Comment (arjun27) [reply docs/debug.md:54]: from what I can tell, vscode 1.46+ does not require a special terminal

**Pull request merged**